File: src/main/java/com/bdev/hogwarts_api/rest_service/public_events/PublicEventsRestServiceImpl.java
package com.bdev.hogwarts_api.rest_service.public_events;

import com.bdev.hogwarts_api.data.dto.events.Event;
import com.bdev.hogwarts_api.data.dto.events.EventParticipant;
import com.bdev.hogwarts_api.data.dto.events.EventType;
import com.bdev.hogwarts_api.exceptions.http.HttpEntityNotFoundException;
import com.bdev.hogwarts_api.service.event_participant.EventParticipantService;
import com.bdev.hogwarts_api.service.events.EventService;
import com.bdev.hogwarts_api.vk.VkMessageSender;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.Arrays;
import java.util.List;

@Service
public class PublicEventsRestServiceImpl implements PublicEventsRestService {
    @Autowired
    private EventService eventService;
    @Autowired
    private EventParticipantService eventParticipantService;

    @Override
    @Transactional(readOnly = true)
    public Event getCurrentEvent(EventType eventType) {
        return eventService
                .getLatestEvent(eventType)
                .orElseThrow(() -> new HttpEntityNotFoundException(""));
    }

    @Override
    @Transactional
    public void addEventParticipant(EventParticipant eventParticipant) {
        if (!eventService.exists(eventParticipant.getEventId())) {
            throw new HttpEntityNotFoundException("");
        }

        eventParticipantService.createParticipant(eventParticipant);

        Event event = eventService
                .getById(eventParticipant.getEventId())
                .orElseThrow(() -> new HttpEntityNotFoundException(""));

        String message =
                "Новая заявка на " + event.getName() + ".\n" +
                "Студент: " + eventParticipant.getName() + "\n" +
                "Телефон: " + eventParticipant.getPhone() + "\n" +
                "VK: ?\n" +
                "Подробнее: http://hogwarts-engschool.ru/admin/#/events/" + event.getId() + "/participants";

        List<Integer> adminIds = Arrays.asList(23236615, 30026296);

        adminIds.forEach(adminId -> new VkMessageSender().sendMessage(adminId, message));
    }
}

File: src/main/java/com/bdev/hogwarts_api/rest_service/public_events/PublicEventsRestServiceImpl.kt
package com.bdev.hogwarts_api.rest_service.public_events

import com.bdev.hogwarts_api.data.dto.events.Event
import com.bdev.hogwarts_api.data.dto.events.EventParticipant
import com.bdev.hogwarts_api.data.dto.events.EventType
import com.bdev.hogwarts_api.exceptions.http.HttpEntityNotFoundException
import com.bdev.hogwarts_api.service.event_participant.EventParticipantService
import com.bdev.hogwarts_api.service.events.EventService
import com.bdev.hogwarts_api.vk.VkMessageSender
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

import java.util.Arrays

@Service
open class PublicEventsRestServiceImpl : PublicEventsRestService {
    @Autowired
    private lateinit var eventService: EventService
    @Autowired
    private lateinit var eventParticipantService: EventParticipantService

    @Transactional(readOnly = true)
    override fun getCurrentEvent(eventType: EventType): Event {
        return eventService
                .getLatestEvent(eventType)
                ?: throw HttpEntityNotFoundException("")
    }

    @Transactional
    override fun addEventParticipant(eventParticipant: EventParticipant) {
        if (!eventService.exists(eventParticipant.eventId)) {
            throw HttpEntityNotFoundException("")
        }

        eventParticipantService.createParticipant(eventParticipant)

        val event = eventService
                .getById(eventParticipant.eventId)
                ?: throw HttpEntityNotFoundException("")

        val message = "Новая заявка на " + event.name + ".\n" +
                "Студент: " + eventParticipant.name + "\n" +
                "Телефон: " + eventParticipant.phone + "\n" +
                "VK: ?\n" +
                "Подробнее: http://hogwarts-engschool.ru/admin/#/events/" + event.id + "/participants"

        val adminIds = Arrays.asList(23236615, 30026296)

        adminIds.forEach { adminId -> VkMessageSender().sendMessage(adminId!!, message) }
    }
}
